### PR TITLE
remove metadata.creationTimestamp field from configMapGenerator

### PR DESCRIPTION
**Pull request by liuhuiping2013**
for this issue:
#344

### CODE REVIEW

**Comment by liuhuiping2013** [comment]
need help, my email is right, why “needs CLA”
what's wrong?

**Comment by Liujingfang1** [comment]
@liuhuiping2013 You need to follow the steps on this link to sign up the CLA https://identity.linuxfoundation.org/projects/cncf

File: pkg/configmapandsecret/configmapfactory.go [hunk 1]
```diff
@@ -70,6 +71,7 @@ func objectToUnstructured(in runtime.Object) (*unstructured.Unstructured, error)
 	}
 	var out unstructured.Unstructured
 	err = out.UnmarshalJSON(marshaled)
+	out.SetCreationTimestamp(metav1.Time{})
```
Comment (Liujingfang1) [inline pkg/configmapandsecret/configmapfactory.go]: `objectToUnstructured` might not be the best place to add the line of setting the creationTimestamp. Can you move this line to functions `MakeConfigMap` and `MakeSecret`?

File: pkg/resource/resource.go [hunk 2]
```diff
@@ -43,6 +44,8 @@ func NewResourceWithBehavior(obj runtime.Object, b GenerationBehavior) (*Resourc
 	}
 	var u unstructured.Unstructured
 	err = u.UnmarshalJSON(marshaled)
+	// creationTimestamp always 'null', remove it
+	u.SetCreationTimestamp(metav1.Time{})
```
Comment (Liujingfang1) [inline pkg/resource/resource.go]: This line can be added to `MakeConfigMap` and `MakeSecet` instead.

Comment (liuhuiping2013) [reply pkg/resource/resource.go]: oh, in metav1.ObjectMeta's definition:
> CreationTimestamp Time `json:"creationTimestamp,omitempty" protobuf:"bytes,8,opt,name=creationTimestamp"`
> DeletionTimestamp *Time `json:"deletionTimestamp,omitempty" protobuf:"bytes,9,opt,name=deletionTimestamp"`
creationTimestamp always non-empty, can not remove it like deleteTimestamp.

this is why i considered remove it from unstructured.Unstructured.

**Comment by liuhuiping2013** [comment]
does "git config user.name" required equal to my github's username?
or "git config user.name" required equal to my github profile's name?
or with no limit for git's user.name, just make sure user.email OK?

**Comment by Liujingfang1** [comment]
@liuhuiping2013 I checked my git config. I only have user.email set.

**Pull request merged**